app/src/main/java/com/duckduckgo/app/widget/AddWidgetLauncher.kt

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2022 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.widget
+
+import android.annotation.SuppressLint
+import android.app.Activity
+import android.app.ActivityOptions
+import android.app.PendingIntent
+import android.appwidget.AppWidgetManager
+import android.content.ComponentName
+import android.content.Context
+import android.content.Intent
+import com.duckduckgo.app.widget.ui.AddWidgetInstructionsActivity
+import com.duckduckgo.app.widget.ui.WidgetCapabilities
+import com.duckduckgo.di.scopes.AppScope
+import com.duckduckgo.widget.SearchAndFavoritesWidget
+import com.squareup.anvil.annotations.ContributesBinding
+import dagger.SingleInstanceIn
+import javax.inject.Inject
+
+interface AddWidgetLauncher {
+    fun launchAddWidget(activity: Activity?)
```

Comment:
Q: Do we need an`Activity` here? I've been checking the code inside the different implementations and I think we only need a context. 

EDIT: if context is what you need. I would encourage you to see if it's possible to automatically inject the context in the concrete class via constructor (but I'm not sure if our setup allows that).

Reply:
Good question. The `AppWidgetManager` doesn't depend at all on Activity object. However, the legacy implementation where `ActivityOptions.makeSceneTransitionAnimation(activity)` specifically requires an Activity. I actually would have wanted to inject the Activity instance to the LegacyAddWidgetLauncher class only, but that would mean that for the BrowserActivity and SettingsActivity component we would need to provide a binding for the Activity like `bindActivity(impl: BrowserActivity): Activity.` which I don't think we currently have.

Reply:
oh, I didn't spot that usage inside `LegacyAddWidgetLauncher`. Sounds reasonable then to keep it like this 👍